src/datasets/summarizations/trend-partial.summarization.ts

```diff
@@ -0,0 +1,103 @@
+import * as math from 'mathjs';
+import { Summary } from './types';
+import { TimeSeriesPoint } from '../metas/types';
+import { cacheSummaries } from './utils/commons';
+import {
+  PointMembershipFunction,
+  MembershipFunction,
+  trapmf,
+  trapmfL,
+  trapmfR,
+} from './libs/protoform';
+import {
+  exponentialMovingAverage,
+  normalizedUniformPartiallyLinearEpsApprox,
+  TimeSeriesPartialTrend,
+} from './libs/trend';
+import { formatX, formatY } from '../../utils/formatters';
+
+export function queryFactory(points: TimeSeriesPoint[]) {
+  return cacheSummaries(() => {
+
+    const smoothedPoints = exponentialMovingAverage(points);
+    const partialTrends = normalizedUniformPartiallyLinearEpsApprox(smoothedPoints, 0.01);
+
+    const applyTrendAngleWithWeight = (f: MembershipFunction) => ({ cone }: TimeSeriesPartialTrend) => {
+      const avgAngleRad = (cone.endAngleRad + cone.startAngleRad) / 2;
+      return f(avgAngleRad);
+    };
+
+    const MAX_ANGLE = Math.atan(500 / 800);

```

Comment:
Add comment for why this is the max angle. I think you had it in another module. You can also have this be a shared constant if it's used across modules